xds/internal/server/listener_wrapper.go

```diff
@@ -341,12 +342,18 @@ func (l *listenerWrapper) Accept() (net.Conn, error) {
 			continue
 		}
 		cw := &connWrapper{Conn: conn, filterChain: fc, parent: l, urc: fc.UsableRouteConfiguration}
-		l.conns = append(l.conns, cw)
-		l.mu.RUnlock()
+		l.conns[cw] = true
+		l.mu.Unlock()
 		return cw, nil
 	}
 }
 
+func (l *listenerWrapper) RemoveConn(conn *connWrapper) {
```

Comment:
Unexport?  I don't think this is here for an interface.

Reply:
Ah good point I forgot conn wrapper was in same package.